### DISCUSSION TITLE
Facing problem in loaded model from colab notebook to flask app

### DISCUSSION THREAD

**Discussion by webstormuser** [mrdbourke/pytorch-deep-learning#707] [Q&A]
Hello team,i have created PetsImageclassificarion model using pytirch in colab notebook and saved `with`  
torch.save(model,path)
And downloade dmodel and loaded using torch.load()in flask app but it started redownloading dataset and retraining model again in flask app in vscode terminal

**Comment by shisirkha** [comment]
Hello! It sounds like you’re experiencing an issue with loading a PyTorch model that you trained in a Colab notebook into a Flask app. Specifically, you mentioned that the model is redownloading the dataset and retraining itself when you load it using torch.load().

One possible reason for this behavior is that the model is not being loaded correctly. When you save a PyTorch model using torch.save(), it saves the model’s state dictionary to a file. To load the model back into memory, you need to create an instance of the model and then load the state dictionary into it using model.load_state_dict(). Here’s an example of how you can do this:



```
import torch
from my_model import MyModel

# Create an instance of the model
model = MyModel()

# Load the state dictionary into the model
state_dict = torch.load('path/to/model.pth')
model.load_state_dict(state_dict)

# Set the model to evaluation mode
model.eval()
```
In this example, we first create an instance of the model using `MyModel()`. Then, we load the state dictionary from the saved file using `torch.load()`. Finally, we load the state dictionary into the model using `model.load_state_dict()` and set the model to evaluation mode using `model.eval()`.

I hope this helps!

**Comment by webstormuser** [comment]
Thank you shisirkha !!

>
>
